src/backend/columnar/columnar_customscan.c

```diff
@@ -486,14 +489,50 @@ static void
 ColumnarScan_ExplainCustomScan(CustomScanState *node, List *ancestors,
 							   ExplainState *es)
 {
-	TableScanDesc scanDesc = node->ss.ss_currentScanDesc;
+	ColumnarScanDesc scanDesc = (ColumnarScanDesc) node->ss.ss_currentScanDesc;
 
 	if (scanDesc != NULL)
 	{
 		int64 chunkGroupsFiltered = ColumnarScanChunkGroupsFiltered(scanDesc);
 		ExplainPropertyInteger("Columnar Chunk Groups Removed by Filter", NULL,
 							   chunkGroupsFiltered, es);
 	}
+
+	char *projectedColumnsStr = GenerateProjectedColumnStr(node->ss);
+	if (projectedColumnsStr)
+	{
+		ExplainPropertyText("Columnar Projected Columns", projectedColumnsStr, es);
+	}
+}
+
+
+/*
+ * GenerateProjectedColumnStr generates projected column string for explain output.
+ * Returns NULL if no columns are projected.
+ */
+static char *
+GenerateProjectedColumnStr(ScanState scanState)
+{
+	Bitmapset *attrNeeded = ColumnarAttrNeeded(&scanState);
+	if (bms_is_empty(attrNeeded))
+	{
+		return NULL;
+	}
+
+	List *columnNameList = NIL;
+
+	int bmsMember = -1;
+	while ((bmsMember = bms_next_member(attrNeeded, bmsMember)) >= 0)
+	{
+		/* ColumnarAttrNeeded decrements attrNumber's by one before storing them in set */
+		AttrNumber attrNumber = bmsMember + 1;
+		Oid relationId = RelationGetRelid(scanState.ss_currentRelation);
+		bool missingOk = false;
+		char *columnName = get_attname(relationId, attrNumber, missingOk);
+		columnNameList = lappend(columnNameList, columnName);
```

Comment:
Also use ruleutils.c:quote_identifier() and have some tests for weird attribute names.